core/controllers/simulationcontroller/simulation_controller.go

```diff
@@ -54,6 +57,33 @@ func (c *SimulationController) WithPrefix(prefix string) *SimulationController {
 	}
 }
 
+func (c *SimulationController) EndSimulation(rw http.ResponseWriter, r *http.Request) {
+	id := mux.Vars(r)["id"]
+	if id == "" {
+		rw.WriteHeader(http.StatusBadRequest)
```

Comment:
I think we should replace this generic 400 response with a 422 Unprocessable Entity.

Actually, if we change the path parameter so that only integer ids are matched, then this error condition will never be hit so we probably don't need this empty string check at all